app/src/main/java/mok/it/app/mokapp/feature/project_add_participants/AddOrRemoveMembersFragment.kt

```diff
@@ -0,0 +1,217 @@
+                SearchField(
+                    searchQuery = searchQuery,
+                    chipState = chipState,
+                    onValueChange = { searchQuery = it },
+                )
+
+                LazyColumn {// selected users
+                    items(users.filter { uiState.selectedUserIds.contains(it.documentId) }) { user ->
+                        UserRow(
+                            user,
+                            true,
+                        ) { _ ->
+                            viewModel.userSelectionClicked(user)
+                        }
+                    }
+                }
+                HorizontalDivider(modifier = Modifier.padding(8.dp))
+                if (filteredUnselectedUsers.isNotEmpty()) {
+                    LazyColumn {// all users
+                        items(filteredUnselectedUsers) { user ->
+                            UserRow(
+                                user,
+                                false,
+                            ) { _ ->
+                                viewModel.userSelectionClicked(user)
+                            }
+                        }
+                    }
+                } else {
+                    Text(
+                        text = "Nincsenek a feltételeknek megfelelő tagok",
+                        modifier = Modifier.padding(16.dp),
+                        style = MaterialTheme.typography.headlineSmall,
+                        textAlign = androidx.compose.ui.text.style.TextAlign.Center
+                    )
+                }
+            }
+        }
+    }
+
+    @Composable
+    fun UserRow(
+        user: User,
+        isSelected: Boolean,
+        onCheckedChange: (Boolean) -> Unit,
+    ) {
+        Row(modifier = Modifier
+            .fillMaxWidth()
+            .height(48.dp)
+            .clickable {
+                onCheckedChange(!isSelected)
+            }
+            .padding(horizontal = 16.dp), verticalAlignment = Alignment.CenterVertically) {
+            UserIcon(user = user, navController = findNavController())
+            Text(
+                text = user.name,
+                style = MaterialTheme.typography.bodyLarge,
+                modifier = Modifier
+                    .padding(start = 16.dp)
+                    .weight(1f)
+            )
+            Checkbox(
+                checked = isSelected,
+                onCheckedChange = null // Accessibility: 'selectable' handles state
+            )
+        }
+    }
+
+    @Composable
+    private fun getFilteredUsers(
+        searchQuery: String,
+        chipState: ChipTextFieldState<Chip>,
+        selectedUserIds: List<String>,
+    ) = viewModel.users.observeAsState().value?.filter { user ->
+        isUserMatched(
+            user, searchQuery, chipState
+        )
+    }.orEmpty()
+        // we put selected users first, then alphabetically
+        .sortedWith(compareBy({ !selectedUserIds.contains(it.documentId) }, { it.name }))
```

Comment:
Kicsit fura ez a compose-ban írt üzleti logika, ezeknek nem lenne jobb helye a VM-ben egy csak gettelhető propertyben? (az isUserMatched metódusra is vonatkozik)

Reply:
de, igen, jogos. Máshol is van ilyen, talán a rewardoknál, azt is megcsinálom akkor